File: src/main/java/com/shreyash/BankApplication/Repository/UserRepository.java
package com.shreyash.BankApplication.Repository;

import com.shreyash.BankApplication.Entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User,Long> {

    Boolean existsByEmail(String email);

    Boolean existsByAccountNumber(String accountNumber);

    User findByAccountNumber(String accountNumber);



}

File: src/main/java/com/shreyash/BankApplication/repository/UserRepository.kt
package com.shreyash.BankApplication.repository

import com.shreyash.BankApplication.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User,Int> {

    fun existsByCustomerID(customerID: String?): Boolean?
    fun findByCustomerID(customerID: String?): User?
}